### PR TITLE
Penalty friendly is applicale and invariant holds test

### LINKED ISSUE #2254: Implement PenaltyFriendly isApplicable and invariantHolds tests
### Description of the task

`isApplicable()` determines if the play should start running
`invariantHolds()` determines if the play should continue running (`isApplicable()` was true earlier to start the play)

Currently, the `isApplicable()` and `invariantHolds()` functions of the play are untested. We had a lot of bugs because of this at Virtual Robocup. We need to implement a test for each.

Take a look at an example of how to do this for `HaltPlay`, the setup of the Play, World, GameState and ASSERT_TRUE/FALSE will be similar for your test: https://github.com/UBC-Thunderbots/Software/pull/2251/files

NOTE: 
* That PR has been heavily commented to make it easier to follow along. You only need to add a couple comments to clarify what your testing, nothing more.
* The createGameState function has been implemented to conviently create GameState from referee commands. You need to poke around `software/world/game_state.cpp` to figure out the right sequence of RefereeCommands. (You should already be slightly familiar with this file from the interview :))

### Acceptance criteria

- [ ] Make sure `isApplicable` and `invariantHolds` are correct: https://robocup-ssl.github.io/ssl-rules/sslrules.html (Rule Book)
- [ ] Implement tests for `isApplicable` and `invariantHolds`

### Blocked By
N/A

### PR DESCRIPTION
**Pull request by divy-07**
### Description

Implemented isApplicable and invariantHolds tests for PenaltyKickPlay. 
isApplicable() determines if the play should start running. invariantHolds() determines if the play should continue running. 

### Testing Done

Unit tests

### Resolved Issues

resolved #2254 

### Length Justification

---

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by PhilD71 — DISMISSED** [review]
LGTM!

File: src/software/ai/hl/stp/play/penalty_kick_play_test.cpp [hunk 1]
```diff
@@ -108,3 +108,29 @@ TEST_F(PenaltyKickPlayTest, DISABLED_test_penalty_kick_take)
...
+    // GameState: The game state to test with.
+    world.updateGameState(::TestUtil::createGameState(
+        RefereeCommand::PREPARE_PENALTY_US, RefereeCommand::PREPARE_PENALTY_US));
+
+    // Lets make sure the play will start running and stay running
+    ASSERT_TRUE(penalty_kick_play.isApplicable(world));
+    ASSERT_TRUE(penalty_kick_play.invariantHolds(world));
+
+    world.updateGameState(::TestUtil::createGameState(
+        RefereeCommand::PREPARE_PENALTY_THEM, RefereeCommand::PREPARE_PENALTY_US));
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/penalty_kick_play_test.cpp:132]: I'd like to see a test with a more natural transition, such as PENALTY_US to NORMAL_START

**Review by jonathanlew — COMMENTED** [review]
just one non-blocking comment

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
seems to be random empty file `...` that made it in

**Comment by divy-07** [comment]
sorry for a lot of commits, I had some issues with git and pushing

File: src/software/ai/hl/stp/play/penalty_kick_play_test.cpp [hunk 2]
```diff
@@ -108,3 +108,29 @@ TEST_F(PenaltyKickPlayTest, DISABLED_test_penalty_kick_take)
...
+    // GameState: The game state to test with.
+    world.updateGameState(::TestUtil::createGameState(
+        RefereeCommand::PREPARE_PENALTY_US, RefereeCommand::PREPARE_PENALTY_US));
+
+    // Lets make sure the play will start running and stay running
+    ASSERT_TRUE(penalty_kick_play.isApplicable(world));
+    ASSERT_TRUE(penalty_kick_play.invariantHolds(world)); 
+
+    world.updateGameState(::TestUtil::createGameState(
+        RefereeCommand::NORMAL_START, RefereeCommand::PREPARE_PENALTY_US));
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/penalty_kick_play_test.cpp]: sorry this wasn't clear, but please also keep the PREPARE_PENALTY_THEM test

**Comment by divy-07** [comment]
> seems to be random empty file `...` that made it in
Fixed!

File: src/software/ai/hl/stp/play/penalty_kick_play_test.cpp [hunk 3]
```diff
@@ -108,3 +108,44 @@ TEST_F(PenaltyKickPlayTest, DISABLED_test_penalty_kick_take)
...
+    world.updateGameState(::TestUtil::createGameState(
+        RefereeCommand::PREPARE_PENALTY_THEM, RefereeCommand::PREPARE_PENALTY_US));
+
+    // Making sure we don't run PenaltyKickPlay
+    ASSERT_FALSE(penalty_kick_play.isApplicable(world));
+    ASSERT_FALSE(penalty_kick_play.invariantHolds(world));
+
+    // test 2: NORMAL_START to PREPARE_PENALTY_US
+    world.updateGameState(::TestUtil::createGameState(RefereeCommand::NORMAL_START,
+                                                      RefereeCommand::NORMAL_START));
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/penalty_kick_play_test.cpp]: nit: this comment doesn't match the code

File: src/software/ai/hl/stp/play/penalty_kick_play_test.cpp [hunk 4]
```diff
@@ -108,3 +108,44 @@ TEST_F(PenaltyKickPlayTest, DISABLED_test_penalty_kick_take)
...
+
+    // World: A blank testing world we will manipulate for the test
+    auto world = ::TestUtil::createBlankTestingWorld();
+
+    // PenaltyKickPlay: the play under test
+    auto penalty_kick_play = PenaltyKickPlay(play_config);
+
+    // Test 1: PREPARE_PENALTY_US to PREPARE_PENALTY_THEM
+    world.updateGameState(::TestUtil::createGameState(
+        RefereeCommand::PREPARE_PENALTY_US, RefereeCommand::PREPARE_PENALTY_US));
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/penalty_kick_play_test.cpp]: nit: this comment doesn't match the code

**Review by jonathanlew — COMMENTED** [review]
just a couple nits

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by akhilveeraghanta — APPROVED** [review]

**Review by PhilD71 — APPROVED** [review]
Good!

**Pull request merged**